cobalt/h5vcc/h5vcc_metrics.cc

```diff
@@ -0,0 +1,49 @@
+// Copyright 2023 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "cobalt/h5vcc/h5vcc_metrics.h"
+
+#include <string>
+
+#include "cobalt/browser/metrics/cobalt_metrics_service_client.h"
+#include "cobalt/browser/metrics/cobalt_metrics_services_manager.h"
+#include "cobalt/h5vcc/h5vcc_metric_type.h"
+#include "cobalt/h5vcc/metric_event_handler_wrapper.h"
+
+namespace cobalt {
+namespace h5vcc {
+
+void H5vccMetrics::OnMetricEvent(
+    const h5vcc::MetricEventHandlerWrapper::ScriptValue& eventHandler) {
+  event_handler_ = new h5vcc::MetricEventHandlerWrapper(this, eventHandler);
+  browser::metrics::CobaltMetricsServiceClient* client =
```

Comment:
Tiny nit, but you may want to DCHECK(client) after this, here and below. If there's any lifetime usage issues you'll get an early warning

Reply:
Done